js/ermrest.js

```diff
@@ -666,18 +685,18 @@ var ERMrest = (function (module) {
             return module._http.get(uri).then(function(response) {
                 return response.data[0].row_count;
             }, function(response) {
-                var error = responseToError(response);
+                var error = module._responseToError(response);
                 return module._q.reject(error);
             });
         },
 
         /**
          *
-         * @param {Object} filter Optional. Negation, Conjunction, Disjunction, UnaryPredicate, BinaryPredicate or null
-         * @param {Number} limit Optional. Number of rows or null
-         * @param {Array} columns Optional. Array of column names or Column objects output
-         * @param {Array} sortby Option. An ordered array of {column, order} where column is column name or Column object, order is null/'' (default), 'asc' or 'desc'
-         * @returns {Promise}
+         * @param {ERMrest.Filters.Negation | ERMrest.Filters.Conjunction | ERMrest.Filters.Disjunction | ERMrest.Filters.UnaryPredicate | ERMrest.Filters.BinaryPredicate} [filter]
+         * @param {Number} [limit] Number of rows
+         * @param {ERMrest.Column[] | string[]} [columns] Array of column names or Column objects output
+         * @param {Object[]} [sortby] An ordered array of {column, order} where column is column name or Column object, order is null (default), 'asc' or 'desc'
+         * @returns {Promise} promise with rowset or error
```

Comment:
is it possible to be more specific about the type of error that might be returned by a rejected promise, or is that unknown in the api? i suppose it might be possible that it could be almost any ermrest error and we can't really say more than this?